lib/spack/spack/package_prefs.py

```diff
@@ -230,7 +230,7 @@ def get_package_dir_permissions(spec):
     attribute sticky for the directory. Package-specific settings take
     precedent over settings for ``all``"""
     perms = get_package_permissions(spec)
-    if perms & stat.S_IRWXG:
+    if perms & stat.S_IRWXG and spack.config.get('config:allow_sgid', True):
```

Comment:
We prefer to have default in *both* the code and the `$spack/etc/spack/defaults` config scope. Could you add

`allow_sgid: true`

to `$spack/etc/spack/defaults/config.yaml` along with a comment explaining that this is for filesystems that don't allow sgid manipulation by unprivileged users?

Reply:
@becker33 done